multicast/src/main/kotlin/com/dropbox/flow/multicast/ChannelManager.kt

```diff
@@ -43,170 +48,253 @@ internal class ChannelManager<T>(
      * Instead, it is kept open and if a new downstream shows up that causes us to restart the flow,
      * it will receive values as well.
      */
-    private val piggybackingDownstream: Boolean = false,
+    piggybackingDownstream: Boolean = false,
     /**
      * Called when a value is dispatched
      */
-    private val onEach: suspend (T) -> Unit,
-    /**
-     * Called when the channel manager is active (e.g. it has downstream collectors and needs a
-     * producer)
-     */
-    private val onActive: (ChannelManager<T>) -> SharedFlowProducer<T>
-) : StoreRealActor<ChannelManager.Message<T>>(scope) {
-    private val buffer = Buffer<T>(bufferSize)
-    /**
-     * The current producer
-     */
-    private var producer: SharedFlowProducer<T>? = null
+    onEach: suspend (T) -> Unit,
 
-    /**
-     * Tracks whether we've ever dispatched value or error from the current producer.
-     * Reset when producer finishes.
-     */
-    private var dispatchedValue: Boolean = false
+    upstream: Flow<T>
+) {
 
-    /**
-     * List of downstream collectors.
-     */
-    private val channels = mutableListOf<ChannelEntry<T>>()
-
-    override suspend fun handle(msg: Message<T>) {
-        when (msg) {
-            is Message.AddChannel -> doAdd(msg)
-            is Message.RemoveChannel -> doRemove(msg.channel)
-            is Message.DispatchValue -> doDispatchValue(msg)
-            is Message.DispatchError -> doDispatchError(msg)
-            is Message.UpstreamFinished -> doHandleUpstreamClose(msg.producer)
-        }
+    private val actor = Actor<T>(scope, bufferSize, piggybackingDownstream, onEach, upstream)
+
+    suspend operator fun plusAssign(channel: SendChannel<Message.Dispatch.Value<T>>) =
+        actor.send(Message.AddChannel(channel))
+
+    suspend operator fun minusAssign(channel: SendChannel<Message.Dispatch.Value<T>>) =
+        actor.send(Message.RemoveChannel(channel))
+
+    suspend fun close() {
+        actor.close()
     }
 
-    /**
-     * We are closing. Do a cleanup on existing channels where we'll close them and also decide
-     * on the list of leftovers.
-     */
-    private fun doHandleUpstreamClose(producer: SharedFlowProducer<T>?) {
-        if (this.producer !== producer) {
-            return
+    private class Actor<T>(
```

Comment:
while receiving the code above, this name confused me as this is not a regular actor but sepcial actor for CM.Can we use a more explicit name for it ?